### PR TITLE
Refactoring the API Logic

### LINKED ISSUE #682: Refactoring the REST API Logic
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**
Redundant code is there which can be merged in single conditional statements

**Expected behavior:**
There should not be any redundant code.

I am working on this.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #682 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Refactored the API logic
-

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ cli/build/
 cli/dist/
 .vscode/
 badgeyay/
+.coverage
```
Comment (gabru-md) [inline .gitignore]: what is `.coverage` @yashLadha  ?
:sweat_smile:

Comment (yashLadha) [reply .gitignore]: When we do the coverage test in the local machine, this is generated. The same tests tests that happen in travis. @gabru-md

Comment (gabru-md) [reply .gitignore]: okay. Thanks :D

**Comment by yashLadha** [comment]
Ping @djmgit . Please review it.

**Comment by gabru-md** [comment]
Can you please give us a way to review the changes. something like a deployment or a screenshot that proves that the changes are working? Because there was a PR during codeheat that solved the Internal Server Error, which was being caused by misplaced `If Else` block. 

If you provide with a way, then we can ensure it :smile: 
I'm just asking because the changes are happening to the `If Else` block :sweat_smile:

**Comment by yashLadha** [comment]
I have made change in the logic part just rearranged the blocks and removed
useless calls. Nothing to the logic. Also as the tests are passing, ensures
that the backend is working.

**Comment by gabru-md** [comment]
If that is the case, then you're right.

Thank You for you PR :)

**Pull request merged**